index.bs

```diff
@@ -2495,21 +4225,79 @@ partial interface MLGraphBuilder {
                 optional MLPadOptions options = {});
 };
 </script>
-<div algorithm=pad>
+
+{{MLPadOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLPadOptions>
+    : <dfn>mode</dfn>
+    ::
+        An {{MLPaddingMode}} [=string=].
+        Specifies the different ways to pad the tensor.
+        The default value is `"constant"`.
+
+    : <dfn>value</dfn>
+    ::
+        A {{float}}.
+        Specifies the padding value when {{MLPadOptions/mode}} is set to `"constant"`.
+        The default value is `0`.
+</dl>
+
+<div>
     **Arguments:**
         - *input*: an {{MLOperand}}. The input tensor.
         - *beginningPadding*: a sequence of {{unsigned long}}. The sequence of unsigned integer values indicating the number of padding values to add at the beginning of each input dimension, of length *N* where *N* is the rank of the input tensor. For each dimension *d* of *input*, *beginningPadding[d]* indicates how many values to add before the content in that dimension.
         - *endingPadding*: a sequence of {{unsigned long}}. The sequence of unsigned integer values indicating the number of padding values to add at the ending of each input dimension, of length *N* where *N* is the rank of the input tensor. For each dimension *d* of *input*, *endingPadding[d]* indicates how many values to add after the content in that dimension.
         - *options*: an optional {{MLPadOptions}}. The optional parameters of the operation.
-            - *mode*: an {{MLPaddingMode}}. The different ways to pad the tensor. When not set, it's assumed to be "constant".
-            - *value*: a {{float}}. The pad value when the *options.mode* is set to *"constant"*. When not set, it's assumed to be 0.
 
     **Returns:** an {{MLOperand}}. The padded output tensor. Each dimension of the output tensor can be calculated as follow:
 
     *output size = beginning padding + input size + ending padding*
+</div>
 
-    <div class="example">
-    <pre highlight="js">
+<details open>
+  <summary>
+    To <dfn dfn-for=MLGraphBuilder>calculate padding output sizes</dfn>, given |input|, |beginningPadding| and |endingPadding|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. Let |shape| be a copy of |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}.
+    1. For |index| between `0` and the rank of |shape|:
+        1. Add to |shape|[|index|] the value of |beginningPadding|[|index|].
+        1. Add to |shape|[|index|] the value of |endingPadding|[|index|].
+    1. Return |shape|.
+  </div>
+</details>
+
+<details open>
+  <summary>
+    The {{MLGraphBuilder/pad(input, beginningPadding, endingPadding, options)}} steps are:
+  </summary>
+  <div algorithm=pad class=algorithm-steps>
+    1. [=Assert=]: the type of |input| is {{MLOperand}}.
+    1. If |beginningPadding| or |endingPadding| is not a sequence of {{unsigned long}}, then [=exception/throw=] a {{TypeError}} and stop.
```

Comment:
Consider add check whether the length of beginningPadding and endingPadding is the same as the length of input tensor.